### ISSUE TITLE
`uv add` git dependency using ssh protocol

### ISSUE THREAD

**Issue by theelderbeever** [astral-sh/uv#9060]
The docs [here](https://docs.astral.sh/uv/pip/packages/#installing-a-package) suggest, accurately, that git repo dependencies are able to to be added as a project dependency by `uv add git+https://path.to.project/repo.git` which works correctly. What is the proper syntax for adding via ssh? Our private repos don't allow https. Everything I have tried so far. says the file doesn't end in a known extension or that the the path isn't a repo.

TLDR: How do I `uv add` with an ssh git path instead of https?

**Comment by theelderbeever** [comment]
@FishAlchemist Not sure how I missed that. Regardless that just gives a public key rejection. Which I have this repo checked out via ssh so that shouldn't be the case. Any idea what `uv` uses for the public key?

**Comment by FishAlchemist** [comment]
@theelderbeever 
I'm unsure about the specific public key used by UV, but it seems that UV simply calls the Git CLI when working with Git repositories. Therefore, is it possible that your SSH authentication has passed, but you're unable to clone the project using Git over SSH?

I recall that uv switched to using the Git CLI for some reason. I don't think uv is switched back to using an embedded one yet.

**Comment by theelderbeever** [comment]
So I think that is the stange thing... I _CAN_ clone the project with the default ssh url that github generates ie: `git@github.com:<organization>/<repository>.git`. But if you try and add any form of `ssh://` to the urls in anyway I will get unauthenticated.

**Comment by charliermarsh** [comment]
Yeah I think you want `uv add git+ssh://git@github.com/...`. We just call through to the Git CLI, so it's possible that there's something wrong with your Git authentication itself?

**Comment by theelderbeever** [comment]
@charliermarsh After some digging I think I figured it out...

`git+ssh://git@github.com/...` - requires READ and WRITE access on the repo being added
`git clone git@github.com/...` - only needs READ access to the repo

From the standpoint of being able to depend on ssh access only repositories where you may want to maintain fine grained access to the repo this seems like a limitation of `uv`. Any ideas here? I wouldn't necessarily say this is closed unless it should be a different issue instead?
